src/vector/app.tsx

```diff
@@ -153,6 +154,26 @@ export async function loadApp(fragParams: {}) {
 
     // Don't bother loading the app until the config is verified
     const config = await verifyServerConfig();
+
+    // Before we continue, let's see if we're supposed to do an SSO redirect
+    const [userId] = await Lifecycle.getStoredSessionOwner();
+    const hasPossibleToken = !!userId;
+    const isReturningFromSso = !!params.loginToken;
+    const autoRedirect = config['sso_immediate_redirect'] === true;
+    if (!hasPossibleToken && !isReturningFromSso && autoRedirect) {
+        console.log("Bypassing app load to redirect to SSO");
+        const tempCli = createClient({
+            baseUrl: config['validated_server_config'].hsUrl,
+            idBaseUrl: config['validated_server_config'].isUrl,
+        });
+        PlatformPeg.get().startSingleSignOn(tempCli, "sso", `/${getScreenFromLocation(window.location).screen}`);
```

Comment:
in the case of the server not being configured with sso this will likely fall apart and leave the user stranded.

With Synapse they'll land on a JSON response of 

```
{
    "errcode": "M_UNKNOWN",
    "error": "SSO requires a valid public_baseurl"
}
```

or similar and using browser back would just send them back here again without any meaningful error

Reply:
correct - this is documented on the config flag as "don't".